lib/elements/dom-repeat.html

```diff
@@ -392,22 +392,26 @@
       return this.__dataHost._methodHost || this.__dataHost;
     }
 
-    __sortChanged(sort) {
-      let methodHost = this.__getMethodHost();
-      this.__sortFn = sort && (typeof sort == 'function' ? sort :
-        function() { return methodHost[sort].apply(methodHost, arguments); });
-      if (this.items) {
-        this.__debounceRender(this.__render);
+    __sortChanged(fn) {
+      if (typeof fn === 'string') {
+        let methodName = fn, obj = this.__getMethodHost();
+        fn = function() { return obj[methodName].apply(obj, arguments); };
       }
+
+      this.__sortFn = fn;
+
+      if (this.items) { this.__debounceRender(this.__render); }
     }
 
-    __filterChanged(filter) {
-      let methodHost = this.__getMethodHost();
-      this.__filterFn = filter && (typeof filter == 'function' ? filter :
-        function() { return methodHost[filter].apply(methodHost, arguments); });
-      if (this.items) {
-        this.__debounceRender(this.__render);
+    __filterChanged(fn) {
+      if (typeof fn === 'string') {
+        let methodName = fn, obj = this.__getMethodHost();
```

Comment:
`const`?

Reply:
I totally agree. This should be const. However, `const` makes the closure compiler scream:
```
polymer.html_script_36.js:381: WARNING - All constants must be typed. The compiler could not infer the type of constant obj. Please use an explicit type annotation. For more information, see:
https://github.com/google/closure-compiler/wiki/Using-NTI-(new-type-inference)#warnings-about-uninferred-constants
        const methodName = fn, obj = this.__getMethodHost();
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
The same warning of course also for `methodName`